Create get-started-ubuntu-instructions.md (#16466)
dotnet/docs: docs/spark/tutorials/get-started-ubuntu-instructions.md

---

> title: Get Started with .NET for Apache Spark on Ubuntu

Reviewer: ```suggestion
title: Get started with .NET for Apache Spark on Ubuntu
```

> * Write a simple .NET for Apache Spark application
> * Build and run your .NET for Apache Spark application

## Pre-requisites

- Download and install the following: **[.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.1)** | **[OpenJDK 8](https://openjdk.java.net/install/)** | **[Apache Spark 2.4.1](https://archive.apache.org/dist/spark/spark-2.4.1/spark-2.4.1-bin-hadoop2.7.tgz)**
- Download and install **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release:
  - Select a **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release from .NET for Apache Spark GitHub Releases page and download into your local machine (e.g., `~/bin/Microsoft.Spark.Worker`).
  - **IMPORTANT** Create a [new environment variable](https://help.ubuntu.com/community/EnvironmentVariables) `DOTNET_WORKER_DIR` and set it to the directory where you downloaded and extracted the Microsoft.Spark.Worker (e.g., `~/bin/Microsoft.Spark.Worker`).

For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).

## Authoring a .NET for Apache Spark App

> - Use the `dotnet` CLI to create a console application.

Reviewer: ```suggestion
1. Use the .NET Core CLI to create a console application.
```

>    ```shell

Reviewer: ```suggestion
   ```dotnetcli
```

   dotnet new console -o HelloSpark
   ```

> - Install `Microsoft.Spark` Nuget package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package)

Reviewer: ```suggestion
1. Install the `Microsoft.Spark` NuGet package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package).
```

    ```shell
    cd HelloSpark
    dotnet add package Microsoft.Spark
    ```

> - Replace the contents of the `Program.cs` file with the following code:

Reviewer: ```suggestion
1. Replace the contents of the *Program.cs* file with the following code:
```

    ```csharp
    using Microsoft.Spark.Sql;

    namespace HelloSpark
    {
        class Program
        {
            static void Main(string[] args)
            {
                var spark = SparkSession.Builder().GetOrCreate();
                var df = spark.Read().Json("people.json");

>                 df.Show();

Reviewer: Same as noted in the two other PRs.

Author: Changed. Thanks!

            }
        }
    }
    ```

> - Use the `dotnet` CLI to build the application:

Reviewer: ```suggestion
1. Use the .NET Core CLI to build the application:
```

>     ```shell

Reviewer: ```suggestion
    ```dotnetcli
```

    dotnet build
    ```

## Running your .NET for Apache Spark App

> - Open your terminal and navigate into your app folder.

Reviewer: ```suggestion
1. Open your terminal and navigate into your app folder.
```

    ```shell
    cd <your-app-output-directory>
    ```

> - Create `people.json` with the following content:

Reviewer: ```suggestion
1. Create *people.json* with the following content:
```

    ```json
    {"name":"Michael"}
    {"name":"Andy", "age":30}
    {"name":"Justin", "age":19}
    ```

> - Run your app.

Reviewer: ```suggestion
1. Run your app.
```

    ```shell
    spark-submit \
    --class org.apache.spark.deploy.dotnet.DotnetRunner \
    --master local \
    microsoft-spark-2.4.x-<version>.jar \
    dotnet HelloSpark.dll
    ```

>     **Note**: This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (e.g., `~/spark/bin/spark-submit`). For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).

Reviewer: ```suggestion
    > [!NOTE]
    > This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (e.g., `~/spark/bin/spark-submit`). For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).

```

> - The output of the application should look similar to the output below:

Reviewer: ```suggestion
1. The output of the application should look similar to the output below:
```

---
title: Get started with .NET for Apache Spark on Ubuntu
description: Discover how to run a .NET for Apache Spark app using .NET Core on Ubuntu.
ms.date: 01/22/2019
ms.topic: tutorial
ms.custom: mvc
# Customer intent: As a developer, I want to write a simple custom application using .NET for Apache Spark on Ubuntu.
---

> # Tutorial: Get Started with .NET for Apache Spark on Ubuntu

Reviewer: ```suggestion
# Tutorial: Get started with .NET for Apache Spark on Ubuntu
```

This tutorial will show you how to run a .NET for Apache Spark app using .NET Core on Ubuntu 18.04.

> [!div class="checklist"]
>
> * Pre-requisites to run your .NET for Apache Spark application
> * Write a simple .NET for Apache Spark application
> * Build and run your .NET for Apache Spark application

## Pre-requisites

- Download and install the following: **[.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1)** | **[OpenJDK 8](https://openjdk.java.net/install/)** | **[Apache Spark 2.4.1](https://archive.apache.org/dist/spark/spark-2.4.1/spark-2.4.1-bin-hadoop2.7.tgz)**
- Download and install **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release:

>   - Select a **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release from .NET for Apache Spark GitHub Releases page and download into your local machine (e.g., `~/bin/Microsoft.Spark.Worker`).

Reviewer: ```suggestion
  - Select a **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release from .NET for Apache Spark GitHub Releases page and download into your local machine (for example, *~/bin/Microsoft.Spark.Worker*).
```

>   - **IMPORTANT** Create a [new environment variable](https://help.ubuntu.com/community/EnvironmentVariables) `DOTNET_WORKER_DIR` and set it to the directory where you downloaded and extracted the Microsoft.Spark.Worker (e.g., `~/bin/Microsoft.Spark.Worker`).

Reviewer: ```suggestion
  - **IMPORTANT** Create a [new environment variable](https://help.ubuntu.com/community/EnvironmentVariables) `DOTNET_WORKER_DIR` and set it to the directory where you downloaded and extracted the Microsoft.Spark.Worker (for example, *~/bin/Microsoft.Spark.Worker*).
```

For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).

> ## Authoring a .NET for Apache Spark App

Reviewer: ```suggestion
## Author a .NET for Apache Spark app
```

1. Use the .NET Core CLI to create a console application.

   ```dotnetcli
   dotnet new console -o HelloSpark
   ```

> 1. Install the `Microsoft.Spark` NuGet package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package).

Reviewer: ```suggestion
1. Install the `Microsoft.Spark` NuGet package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/nuget/consume-packages/ways-to-install-a-package).
```

>     ```shell

Reviewer: ```suggestion
    ```dotnetcli
```

    cd HelloSpark
    dotnet add package Microsoft.Spark
    ```

1. Replace the contents of the *Program.cs* file with the following code:

    ```csharp
    using Microsoft.Spark.Sql;

    namespace HelloSpark
    {
        class Program
        {
            static void Main(string[] args)
            {
                var spark = SparkSession.Builder().GetOrCreate();
                DataFrame df = spark.Read().Json("people.json");
                df.Show();
            }
        }
    }
    ```

1. Use the .NET Core CLI to build the application:

    ```dotnetcli
    dotnet build
    ```

## Running your .NET for Apache Spark App

1. Open your terminal and navigate into your app folder.

    ```shell
    cd <your-app-output-directory>
    ```

1. Create *people.json* with the following content:

    ```json
    {"name":"Michael"}
    {"name":"Andy", "age":30}
    {"name":"Justin", "age":19}
    ```

1. Run your app.

    ```shell
    spark-submit \
    --class org.apache.spark.deploy.dotnet.DotnetRunner \
    --master local \
    microsoft-spark-2.4.x-<version>.jar \
    dotnet HelloSpark.dll
    ```

    > [!NOTE]

>     > This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (e.g., `~/spark/bin/spark-submit`). For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).

Reviewer: ```suggestion
    > This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (for example, *~/spark/bin/spark-submit*). For detailed instructions, you can see [Building .NET for Apache Spark from Source on Ubuntu](../building/ubuntu-instructions.md).
```

1. The output of the application should look similar to the output below:

    ```text
    +----+-------+
    | age|   name|
    +----+-------+
    |null|Michael|
    |  30|   Andy|
    |  19| Justin|
    +----+-------+
    ```
    
## Next steps

> In this tutorial, you learn how to run your .NET for Apache Spark on Ubuntu. Continue to learn more about how to run your .NET for Apache Spark on Windows and MacOSX.

Reviewer: ```suggestion
In this tutorial, you learn how to run your .NET for Apache Spark on Ubuntu. Continue to learn more about how to run your .NET for Apache Spark on Windows and macOS.
```

> [!div class="nextstepaction"]

> > [Tutorial: Get Started with .NET for Apache Spark on Windows](get-started-windows-instructions.md)

Reviewer: ```suggestion
> [Tutorial: Get started with .NET for Apache Spark on Windows](get-started-windows-instructions.md)
```

> > [Tutorial: Get Started with .NET for Apache Spark on MacOSX](get-started-macos-instructions.md)

Reviewer: ```suggestion
> [Tutorial: Get started with .NET for Apache Spark on macOS](get-started-macos-instructions.md)
```

---
title: Get started with .NET for Apache Spark on Ubuntu
description: Discover how to run a .NET for Apache Spark app using .NET Core on Ubuntu.

> ms.date: 01/22/2019

Reviewer: ```suggestion
ms.date: 01/22/2020
```
